tests/Integration/Classes/ObjectModelTest.php

```diff
@@ -0,0 +1,691 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Classes;
+
+use Configuration;
+use Db;
+use Language;
+use ObjectModel;
+use PHPUnit\Framework\TestCase;
+use Shop;
+
+class ObjectModelTest extends TestCase
+{
+    private const DEFAULT_LANGUAGE_PLACEHOLDER = 'default_language';
+    private const SECOND_LANGUAGE_PLACEHOLDER = 'second_language';
+
+    private const DEFAULT_SHOP_PLACEHOLDER = 'default_shop';
+    private const SECOND_SHOP_PLACEHOLDER = 'second_shop';
+
+    /**
+     * @var int
+     */
+    private $defaultLanguageId;
+
+    /**
+     * @var int
+     */
+    private $secondLanguageId;
+
+    /**
+     * @var int
+     */
+    private $defaultShopId;
+
+    /**
+     * @var int
+     */
+    private $secondShopId;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->installTestableObjectTables();
+        $this->installLanguages();
+        $this->installShops();
+    }
+
+    private function installTestableObjectTables(): void
+    {
+        $testableObjectSqlFile = dirname(__DIR__, 2) . '/Resources/sql/install_testable_object.sql';
+        $sqlRequest = file_get_contents($testableObjectSqlFile);
+        $sqlRequest = preg_replace('/PREFIX_/', _DB_PREFIX_, $sqlRequest);
+
+        $dbCollation = Db::getInstance()->getValue('SELECT @@collation_database');
+        $allowedCollations = ['utf8mb4_general_ci', 'utf8mb4_unicode_ci'];
```

Comment:
I think you can avoid these preg_replace and directly put the desired behavior in the sql file :)

Reply:
Probably, but I prefer tests to remain a bit dynamic just in case, it's not that complicated nor performance taxing anyway